tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/cleaning_tests_base.py

```diff
@@ -0,0 +1,224 @@
+"""
+Integration Test for the rdr_observation_source_concept_id_suppression module.
+
+Remove three irrelevant observation_source_concept_ids from the RDR dataset.
+
+Original Issue:  DC-734 implements integration tests for DC-529
+
+The intent is to remove PPI records from the observation table in the RDR
+export where observation_source_concept_id in (43530490, 43528818, 43530333).
+The records for removal should be archived in the dataset sandbox.  It should
+also ensure that records that have null values or do not match the specified
+ids are not removed.
+"""
+# Python imports
+import unittest
+
+# Third party imports
+import google.cloud.exceptions as gc_exc
+
+# Project imports
+from cdr_cleaner import clean_cdr_engine as engine
+from utils import bq
+
+
+class BaseTest:
+
+    class CleaningRulesTestBase(unittest.TestCase):
+        """
+        A base class for cleaning rules integration tests.
+
+        This class implements setUp, setUpClass, tearDown, and tearDownClass
+        methods that can be inherited and used by all extending classes.  This
+        allows developers to focus more fully on creating a working test rather
+        than setting up or tearing down their test environment.
+        """
+
+        # this should always be for a test project.  a check is implemented to
+        # make sure this is true
+        project_id = ''
+        # a list of test dataset(s) that mimic the dataset(s) this rule is
+        # intended to clean.  this is required for data loading
+        test_dataset_ids = []
+        # a list of dataset(s) that hold sandboxed table(s) created by the
+        # cleaning rules.   required for rule# execution.  a sandbox table
+        # can only be made in an existing dataset.
+        sandbox_dataset_ids = []
+        # a list of fully qualified table names the cleaning rule is targeting.
+        # fq = {project_id}.{dataset_id}.{table_name}.  required for
+        # test setup and cleanup.
+        fq_table_names = []
+        # a list of fully qualified sandbox tables the rule may create.
+        # required for cleanup
+        fq_sandbox_table_names = []
+        # a list of dictionaries where each dictionary defines table
+        # information for one table upon rule execution, such as the
+        # 'name', 'fq_table_name', 'fq_sanbox_table_name', 'loaded_ids',
+        # 'sandboxed_ids', and 'cleaned_ids'
+        tables_and_counts = []
```

Comment:
This has been moved to the extending class `DropRowsTestBase`.  It has been better documented.  It is not used for a single assert statement.  It uses the values here for multiple pre-test assert statements and multiple post test assert statements.  I think it is easier to identify this information like this, rather than having to create multiple assert statements in its place.  The `ObservationSourceConceptIDRowSuppressionTest` that is extending the `DropRowsTestBase` class is only operating on a single table.  However, there are multiple clean rules that operate on multiple tables.  For example, the `EnsureDateDatetimeConsistency` class creates multiple rules that do the same thing but target different tables with possibly different fields.